react-hot-button-component/my-app/src/App.js

```diff
@@ -0,0 +1,39 @@
+import './App.css';
+import HotButton from './HotButton';
+import { useState } from 'react';
+
+function App() {
+  let [temp, setTemp] = useState('dark');
```

Comment:
Having a `temp` for state is completely fine. However, if possible we could have determined the className using the value for `counter`. 

Your approach is still fine, but when it is possible it could be argued creating less state variables is better. 

This is a possible rewrite:
```function App() {
  let [counter, setCounter] = useState(0);

  function handleClick() {
    setCounter(counter + 1);
  }

  let temp;
  if (counter < 2) {
    temp = 'dark';
  } else if (counter < 5) {
    temp = 'cold';
  } else if (counter < 8) {
    temp = 'cool';
  } else if (counter < 11) {
    temp = 'tepid';
  } else if (counter < 14) {
    temp = 'warm';
  } else if (counter < 17) {
    temp = 'hot';
  } else {
    temp = 'nuclear';
  }

  return (
    <div className="App">
      <header className="App-header">
        <HotButton onCustomClick={handleClick} classTemp={temp}></HotButton>
        <p>Clicks: {counter}</p>
      </header>
    </div>
  );
}

export default App;
```